zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/AuthorizationControllerTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.CamundaServiceException;
+import io.camunda.service.security.auth.Authentication;
+import io.camunda.zeebe.gateway.rest.RestControllerTest;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.AuthorizationController;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.dto.AuthorizationAssignRequest;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+
+@WebMvcTest(AuthorizationController.class)
+public class AuthorizationControllerTest extends RestControllerTest {
+  @MockBean private AuthorizationServices<AuthorizationRecord> authorizationServices;
+
+  @BeforeEach
+  void setup() {
+    when(authorizationServices.withAuthentication(any(Authentication.class)))
+        .thenReturn(authorizationServices);
+  }
+
+  @Test
+  void createAuthorizationShouldReturnNoContent() {
+    final var request = new AuthorizationAssignRequest();
+    request.setOwnerKey("1");
+    request.setOwnerType(AuthorizationOwnerType.USER);
+    request.setResourceKey("2");
+    request.setResourceType("resourceType");
+    request.setPermissions(List.of("permission1", "permission2"));
+
+    final var authorizationRecord = new AuthorizationRecord();
+    authorizationRecord.setOwnerKey(request.getOwnerKey());
+    authorizationRecord.setOwnerType(request.getOwnerType());
+    authorizationRecord.setResourceKey(request.getResourceKey());
+    authorizationRecord.setResourceType(request.getResourceType());
+    authorizationRecord.setPermissions(request.getPermissions());
```

Comment:
```suggestion
    final var authorizationRecord = new AuthorizationRecord()
    .setOwnerKey(request.getOwnerKey())
    .setOwnerType(request.getOwnerType())
    .setResourceKey(request.getResourceKey())
    .setResourceType(request.getResourceType())
    .setPermissions(request.getPermissions());
```

🔧 Let's chain these method calls